addons/mrp/models/mrp_workorder.py

```diff
@@ -650,29 +650,35 @@ def button_start(self, raise_on_invalid_state=False):
     def button_finish(self):
         date_finished = fields.Datetime.now()
         all_vals_dict = defaultdict(lambda: self.env['mrp.workorder'])
+        to_be_ended_workorders = self.env['mrp.workorder']
+        moves = (self.move_raw_ids + self.production_id.move_byproduct_ids.filtered(lambda m: (m.operation_id == self.operation_id)))
```

Comment:
nitpick: external `()` are not necessary. I see that before it was like that also, but since the line is changed anyway, might fix that.